src/daemon/ca/commands.rs

```diff
@@ -160,71 +176,93 @@ impl From<CmdDet> for StorableCaCommand {
+                StorableCaCommand::ChildUpdateResources { child, resources }
             }
-            CmdDet::ChildUpdateResources(child, res) => StorableCaCommand::ChildUpdateResources(child, res),
-            CmdDet::ChildUpdateId(child, id) => StorableCaCommand::ChildUpdateId(child, id.ski_hex()),
+            CmdDet::ChildUpdateId(child, id) => StorableCaCommand::ChildUpdateId {
+                child,
+                ski: id.ski_hex(),
+            },
             CmdDet::ChildCertify(child, req, _, _) => {
-                let (rcn, limit, csr) = req.unpack();
+                let (resource_class_name, limit, csr) = req.unpack();
                 let ki = csr.public_key().key_identifier();
-                StorableCaCommand::ChildCertify(child, rcn, limit, ki)
+                StorableCaCommand::ChildCertify {
+                    child,
+                    resource_class_name,
+                    limit,
+                    ki,
+                }
             }
-            CmdDet::ChildRevokeKey(child, req, _, _) => StorableCaCommand::ChildRevokeKey(child, req),
-            CmdDet::ChildRemove(child, _, _) => StorableCaCommand::ChildRemove(child),
+            CmdDet::ChildRevokeKey(child, revoke_req) => StorableCaCommand::ChildRevokeKey { child, revoke_req },
+            CmdDet::ChildRemove(child) => StorableCaCommand::ChildRemove { child },
 
             // ------------------------------------------------------------
             // Being a child
             // ------------------------------------------------------------
             CmdDet::GenerateNewIdKey(_) => StorableCaCommand::GenerateNewIdKey,
-            CmdDet::AddParent(parent, contact) => StorableCaCommand::AddParent(parent, contact.into()),
-            CmdDet::UpdateParentContact(parent, contact) => {
-                StorableCaCommand::UpdateParentContact(parent, contact.into())
-            }
-            CmdDet::RemoveParent(parent) => StorableCaCommand::RemoveParent(parent),
-            CmdDet::UpdateResourceClasses(parent, entitlements, _) => {
-                let mut classes = BTreeMap::new();
-                for entitlement in entitlements.classes() {
-                    classes.insert(entitlement.class_name().clone(), entitlement.resource_set().clone());
+            CmdDet::AddParent(parent, contact) => StorableCaCommand::AddParent {
+                parent,
+                contact: contact.into(),
+            },
+            CmdDet::UpdateParentContact(parent, contact) => StorableCaCommand::UpdateParentContact {
+                parent,
+                contact: contact.into(),
+            },
+            CmdDet::RemoveParent(parent) => StorableCaCommand::RemoveParent { parent },
+            CmdDet::UpdateEntitlements(parent, cmd_entitlements, _) => {
+                let mut entitlements = vec![];
+                for entitlement in cmd_entitlements.classes() {
+                    entitlements.push(StorableRcEntitlement {
```

Comment:
By switching from a HashMap to a vec here has this made it possible for duplicates to exist where none could exist before?